scripts/install.js

```diff
@@ -49,25 +41,26 @@ function download(url, dest, cb) {
  */
 
 function applyProxy(options, cb) {
-  require('npmconf').load({}, function (er, conf) {
-    var getProxyFromEnv = true;
-    ['https-proxy', 'proxy', 'http-proxy'].forEach(function(setting) {
-      var proxyUrl = conf.get(setting);
-
-      if(proxyUrl && proxyUrl === require('url').parse(proxyUrl)) {
-        options.proxy = proxyUrl;
-        getProxyFromEnv = false;
-        cb(options);
-        return;
-      }
-    });
+  npmconf.load({}, function (er, conf) {
+    var proxyUrl;
+
+    if (!er) {
+      ['https-proxy', 'proxy', 'http-proxy'].some(function(setting) {
```

Comment:
FWIW this could be reduced to 

```js
var proxyUrl = ['https-proxy', 'proxy', 'http-proxy'].filter(function(setting) { return conf.get(setting); })[0];
```